### DISCUSSION TITLE
Next Auth without a database adapter

### DISCUSSION THREAD

**Discussion by saadi925** [nextauthjs/next-auth#11556] [Ideas]
### Goals

1. database agnostic next auth
2. custom backend support for auth providers

### Background

like next auth depends upon the database adapter , what if someone's database is not in next js .

like i have prisma in my nest js project with a great schema , account , user, tokens ,blacklisting etc
i don't want to be redundant having two prisma files and , and like generating types on both sides etc .
and i did not find a solution like using it without a database adapter , my oauth is in nest js with custom strategies and guards.

next auth is nice when you want to have a solution only in nextjs , like it is limited , but if backend is somewhere else , like an api , it is a disaster i guess.

### Proposal

we could write a seperate abstraction for it , or enhanced the current one. if this is to be impleemented , i think providing a sepearte function is better , where we don;t really need to provide things that are related to database.

**Comment by BDDoylan** [comment]
I want to bump this, I am facing the EXACT same issue and the documentation does not cover any of my needs or even any videos online. I have my NestJS app with next-auth and a separate NestJS app with drizzle, and I cannot seem to setup next-auth correctly...
